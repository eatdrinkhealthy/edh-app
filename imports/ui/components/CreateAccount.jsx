// @flow
import React from "react";

const CreateAccount = (): React$Element<*> => (
  <div className="pt2 pl4 pr4 pb4">
    <form className="measure center">
      <fieldset id="sign_up" className="ba b--transparent ph0 mh0">
        <legend className="f4 fw6 ph0 mh0">Create Account</legend>
        <div className="mt3">
          <label className="db fw6 lh-copy f6" htmlFor="username">Username</label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            type="text"
            name="username"
            id="username"
          />
        </div>
        <div className="mt3">
          <label className="db fw6 lh-copy f6" htmlFor="email-address">Email</label>
          <input
            className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            type="email"
            name="email-address"
            id="email-address"
          />
        </div>
        <div className="mv3">
          <label className="db fw6 lh-copy f6" htmlFor="password">Password</label>
          <input
            className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            type="password"
            name="password"
            id="password"
          />
        </div>
        <div className="mv3">
          <label className="db fw6 lh-copy f6" htmlFor="confirm-password">Confirm Password</label>
          <input
            className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
            type="password"
            name="confirm-password"
            id="confirm-password"
          />
        </div>
      </fieldset>
      <div>
        <input
          className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
          type="submit"
          value="Create Account"
        />
      </div>
    </form>
  </div>
);

export default CreateAccount;
