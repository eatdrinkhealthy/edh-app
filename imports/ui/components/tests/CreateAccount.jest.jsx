// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import renderer from "react-test-renderer";
import CreateAccount from "../CreateAccount";
import mountCreateAccountForm from "./CreateAccount_helper";

describe("<CreateAccount />", function () {
  it("matches render snapshot", function () {
    // Per a React blog post, when using renderer, must mock out refs
    // https://facebook.github.io/react/blog/2016/11/16/react-v15.4.0.html#mocking-refs-for-snapshot-testing
    // eslint-disable-next-line flowtype/no-weak-types
    function createNodeMock(element: HTMLElement): ?Object {
      if (element.type === "input") {
        return {
          focus() {},
        };
      }
      return null;
    }
    const options = { createNodeMock };

    const tree = renderer.create(
      <CreateAccount handleSubmit={() => {}} />,
      options,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call handleSubmit on submit with form field values", function () {
    const props = {
      handleSubmit: jest.fn(),
    };

    const wrapper = mountCreateAccountForm(
      <CreateAccount {...props} />,
      "user12",
      "user12@test.com",
      "user12pw",
      "user12pw",
    );

    wrapper.find("input[type='submit']").simulate("submit");
    expect(props.handleSubmit).toHaveBeenCalledWith("user12", "user12@test.com", "user12pw");
  });

  it("should clear input fields and give username focus on successful submit", function () {
    const props = {
      handleSubmit: jest.fn(),
    };

    const wrapper = mountCreateAccountForm(
      <CreateAccount {...props} />,
      "user12",
      "user12@test.com",
      "user12pw",
      "user12pw",
    );
    const usernameNode = wrapper.find("input#username");
    const confirmPasswordNode = wrapper.find("input#confirmPassword");

    // give focus to confirm password input (like a user would do before submit)
    // $FlowFixMe
    confirmPasswordNode.get(0).focus();
    wrapper.find("input[type='submit']").simulate("submit");

    expect(usernameNode.props().value).toEqual("");
    expect(wrapper.find("input#email").props().value).toEqual("");
    expect(wrapper.find("input#password").props().value).toEqual("");
    expect(confirmPasswordNode.props().value).toEqual("");
    expect(usernameNode.get(0)).toBe(document.activeElement);
  });

  it("should set focus to username input on render", function () {
    const props = {
      handleSubmit: jest.fn(),
    };

    const wrapper = mountCreateAccountForm(
      <CreateAccount {...props} />,
      "user12",
      "user12@test.com",
      "user12pw",
      "user12pw",
    );

    expect(wrapper.find("input#username").get(0)).toBe(document.activeElement);
  });
});
