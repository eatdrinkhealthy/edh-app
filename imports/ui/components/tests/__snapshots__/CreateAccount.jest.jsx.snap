// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateAccount /> matches render snapshot 1`] = `
<div
  className="pt2 pl4 pr4 pb4"
>
  <form
    className="measure center"
    onSubmit={[Function]}
  >
    <fieldset
      className="ba b--transparent ph0 mh0"
      id="sign_up"
    >
      <legend
        className="f4 fw6 ph0 mh0"
      >
        Create Account
      </legend>
      <div
        className="mt3"
      >
        <label
          className="db fw6 lh-copy f6"
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
          id="username"
          name="username"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="mt3"
      >
        <label
          className="db fw6 lh-copy f6"
          htmlFor="email"
        >
          Email
        </label>
        <input
          className="pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />
      </div>
      <div
        className="mv3"
      >
        <label
          className="db fw6 lh-copy f6"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
      <div
        className="mv3"
      >
        <label
          className="db fw6 lh-copy f6"
          htmlFor="confirmPassword"
        >
          Confirm Password
        </label>
        <input
          className="b pa2 input-reset ba bg-transparent hover-bg-black hover-white w-100"
          id="confirmPassword"
          name="confirmPassword"
          onChange={[Function]}
          type="password"
          value=""
        />
      </div>
    </fieldset>
    <div>
      <input
        className="b ph3 pv2 input-reset ba b--black bg-transparent grow pointer f6 dib"
        type="submit"
        value="Create Account"
      />
    </div>
  </form>
</div>
`;
